### PR TITLE
fix paginator reference

### PR DESCRIPTION
**Pull request by tymondesigns**
rework of #448 

targeting master now

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Surely this isn't correct. We should be typhinting an interface here?

**Comment by GrahamCampbell** [comment]
We should be type hinting `Illuminate\Contracts\Pagination\Paginator`.

**Comment by tymondesigns** [comment]
I know what you mean but..

It would have to be `Illuminate\Contracts\Pagination\LengthAwarePagination` since fractal requires that as described in my initial PR. 

And as @jasonlewis mentioned, that would remove the possibility of people using the standard `Illuminate\Pagination\Pagination` implementation.

Right?

**Comment by jasonlewis** [comment]
I actually forgot about the interface. That would be better to type hint then the abstract. My bad.

**Comment by tymondesigns** [comment]
No worries, Contract it is :+1: 

Might be worth mentioning in the wiki that they will not be able to use `User::simplePaginate()` for example, since that returns a standard Paginator instance.

File: src/Http/Response/Factory.php [hunk 1]
```diff
@@ -7,8 +7,8 @@
 use Illuminate\Support\Str;
 use Dingo\Api\Http\Response;
 use Illuminate\Support\Collection;
-use Illuminate\Pagination\Paginator;
 use Dingo\Api\Transformer\Factory as TransformerFactory;
+use Illuminate\Contracts\Pagination\LengthAwarePaginator;
```
Comment (jasonlewis) [inline src/Http/Response/Factory.php]: Should be just `Paginator`.

I know Fractal needs a `LengthAwarePaginator` instance but that's assuming everyone will use Fractal. Keep in mind I'm not up to speed on all the pagination changes but this type hint seems to be limiting it to a single type.

Comment (tymondesigns) [inline src/Http/Response/Factory.php]: Yea, I agree it does seem restrictive, but if we use `Illuminate\Contracts\Pagination\Paginator` then fractal won't work at all.

To give some context:
```php
$users = User::paginate(); // returns Illuminate\Pagination\LengthAwarePaginator

// will throw an exception
return $this->response()->pagination($users, new UserTransformer);
```

Or am I missing a trick here?

Comment (tymondesigns) [inline src/Http/Response/Factory.php]: Thinking about it again.. maybe Fractal needs two illuminate paginator adapters. One for each paginator type, then at least we could get fractal to work in either case

Comment (jasonlewis) [inline src/Http/Response/Factory.php]: No... It shouldn't as the `LengthAwarePaginator` contract extends the `Paginator` contract. So even if you pass in an instance of `Illuminate\Pagination\LengthAwarePaginator` it should be fine as it is, through inheritance, an instance of `Illuminate\Contracts\Pagination\Paginator`.

Comment (tymondesigns) [inline src/Http/Response/Factory.php]: LOL.. I totally missed that the interface extends the other, Sorry about that.

All good then :+1: 

I will change it to the Paginator contract

**Comment by tymondesigns** [comment]
all done

**Comment by bweston92** [comment]
When will this be merged? Kind of need this now.

**Comment by jasonlewis** [comment]
All merged. :smile:

**Pull request merged**